app/models/products/user_note_mode.rb

```diff
@@ -0,0 +1,74 @@
+module Products
+
+  class UserNoteMode
+
+    VALID_MODES = %w(hidden optional required).freeze
+
+    attr_reader :raw_value
+
+    # An array of UserNoteModes representing the possible modes
+    # Use this in validations and form collections.
+    def self.all
+      VALID_MODES.map { |value| new(value) }
+    end
+
+    def self.from_string(mode_string)
```

Comment:
The main reason for moving it to a class method/factory rather than the initializer it to allow it to return an instance of something other than `UserNoteMode` (e.g. `InvalidUserNoteMode.`, or in an extreme version of this where each mode is actually a separate class the API doesn't need to change). 

I think I'm probably going to end up using `[]` if y'all are cool with it.